src/api/sendgrid.ts

```diff
@@ -0,0 +1,53 @@
+import mail, { MailDataRequired } from '@sendgrid/mail'
+import apiEnv from './apiEnv';
+
+export interface Personalizations {
+    to: [{
+        email: string,
+        name: string
+    }];
+    dynamic_template_data?: Record<string, string>;
+}
+
+mail.setApiKey(apiEnv.SENDGRID_API_KEY);
+
+const sendEmail = async (personalizations: Personalizations[], template_id: string) => {
+
+    console.log("SendGrid Mail Executed");
+
+    const message: MailDataRequired = {
+        personalizations: personalizations,
+        from: 'yizhang1728@gmail.com', // yuanjian@163
+        templateId: template_id,
+    };
+
+    try {
+        const res = await mail.send(message);
+        console.log("SendGrid Mail sent", res);
+    } catch (e) {
```

Comment:
1. don't try, don't catch, don't swallow exceptions. let it propagate to upper levels. simply:

```
await mail.send()
```

with nothing else.

2. do you need to handle the return value for error detection? if not, remove `const res =`.